pkg/types/snapstore.go

```diff
@@ -65,26 +68,41 @@ const (
 type SnapStore interface {
 	// Fetch should open reader for the snapshot file from store.
 	Fetch(Snapshot) (io.ReadCloser, error)
-	// List will return sorted list with all snapshot files on store.
-	List() (SnapList, error)
+	// List returns a sorted list (based on the last revision, ascending) of all snapshots in the store.
+	// includeAll specifies whether to include all snapshots while listing, including those with exclude tags.
+	// Snapshots with exclude tags are not listed unless includeAll is set to true.
+	List(includeAll bool) (SnapList, error)
 	// Save will write the snapshot to store.
 	Save(Snapshot, io.ReadCloser) error
 	// Delete should delete the snapshot file from store.
 	Delete(Snapshot) error
 }
 
-// Snapshot structure represents the metadata of snapshot.s
+// Snapshot structure represents the metadata of snapshot.
 type Snapshot struct {
-	Kind              string    `json:"kind"` //incr:incremental,full:full
+	Kind              string    `json:"kind"` // incr:incremental, full:full
 	StartRevision     int64     `json:"startRevision"`
-	LastRevision      int64     `json:"lastRevision"` //latest revision on snapshot
+	LastRevision      int64     `json:"lastRevision"` // latest revision on snapshot
 	CreatedOn         time.Time `json:"createdOn"`
 	SnapDir           string    `json:"snapDir"`
 	SnapName          string    `json:"snapName"`
 	IsChunk           bool      `json:"isChunk"`
 	Prefix            string    `json:"prefix"`            // Points to correct prefix of a snapshot in snapstore (Required for Backward Compatibility)
-	CompressionSuffix string    `json:"compressionSuffix"` // CompressionSuffix depends on compessionPolicy
+	CompressionSuffix string    `json:"compressionSuffix"` // CompressionSuffix depends on compression policy
 	IsFinal           bool      `json:"isFinal"`
+	RetentionExpiry   time.Time `json:"retentionExpiry"`
```

Comment:
```suggestion
	ObjectLockRetentionExpiry   time.Time `json:"objectLockRetentionExpiry"`
```

Recently I came across this comment: https://github.com/gardener/etcd-backup-restore/pull/776#discussion_r1764632699 from @renormalize and I found that there can be confusion b/w retention period of objects used by `bucketLock` for immutability and retention period term used by backup-restore's garbage collection policy to garbage collect the snapshots. That's why I want to rename this to avoid any ambiguity. wdyt ?

Reply:
@ishan16696
What do you think of `ImmutabilityExpiryTime`?
I am leaning towards using "immutability" as the word everywhere in etcd-backup-restore to refer to WORM behavior, and this makes it more explicit.

Reply:
yeah, make sense to me, it's good enough to change to avoid any ambiguity.

Reply:
Thanks.

Reply:
Addressed in [a6680d4](https://github.com/gardener/etcd-backup-restore/pull/776/commits/a6680d4ad3e2247f52374b5423b2ada4016e6b1c).